azure-pipelines.yml

```diff
@@ -435,7 +443,98 @@ stages:
         name: revertPomChanges
         displayName: Revert changes made to pom.xml to not break cache feature
 
+- stage: promoteNugets
+  displayName: Promote NuGet on repox
+  dependsOn: qa
+  condition: succeeded()
+  jobs:
+    - job: promoteRepox
+      displayName: 'Promote SonarAnalyzer.CFG.CSharp NuGet on repox'
+      steps:
+        - powershell: |
+            # Calculate the file path
+            $versionFilePath = "$env:BUILD_SOURCESDIRECTORY\scripts\version\Version.props"
+            Write-Host "Reading the Sonar project version from '${versionFilePath}' ..."
+
+            # Read the version from the file
+            [xml]$versionProps = Get-Content "$versionFilePath"
+            $sonarProjectVersion = $versionProps.Project.PropertyGroup.MainVersion
+            Write-Host "Sonar project version is '${sonarProjectVersion}'"
+
+            # Set the variable to it can be used by other tasks
+            Write-Host "##vso[task.setvariable variable=SONAR_PROJECT_VERSION;]$sonarProjectVersion"
+          displayName: 'Read Sonar project version from the versions.props file'
+
+        - task: UsePythonVersion@0
+          inputs:
+            versionSpec: '3.x'
+        
+        - bash: pip3 install requests
+
+        - task: PythonScript@0
```

Comment:
Python scripting is used in the template yml files for similar purposes (promotion & burgr notification), so I would keep the same way of doing things here as well, to be consistent and so that anyone (aside from LT.NET) can also update it if necessary.
Powershell script here is used to set project version environment variable, and the same applies, it is already used as is in other stages of the script for this purpose, so to be consistent on this side as well, I kept this part in powershell